### PR TITLE
add cloud build support

### PR DESCRIPTION
**Pull request by samthor**
This adds two Cloud Build tasks: one to check whether we _need_ to deploy, and one to deploy.

This is combined with the config in the `santa-staging` Cloud project (which sets up triggers that can run basically in cron).

Unrelated stuff that I had to change:
* Moved from Yarn to NPM to simplify a bunch of stuff
* Removes the symlink static/node_modules which pointed to the top-level node_modules, instead, we create it in postinstall. I'm not incredibly happy with this (feels fragile?), but Cloud Build refuses to upload it.
(This might only matter when you upload from your machine [testing], not when Cloud Build checks you out [normal process]).

### CODE REVIEW

File: .cloudbuild/staging/appengine/http.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	port := os.Getenv("PORT")
+	if port == "" {
+		port = "8080"
+	}
+	addr := ":" + port
+	log.Printf("Serving on %s...", addr)
+	http.ListenAndServe(addr, nil)
+}
+
+// // stripPrefix removes a matching prefix from the given URL, but also inserts a
```
Comment (Jezzamonn) [inline .cloudbuild/staging/appengine/http.go]: I would generally opt for deleting commented out functions :)

File: .cloudbuild/staging-deploy.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# This Cloud Build task runs the deploy to staging.
+
+steps:
+  - name: node
+    id: 'Install dependencies'
+    entrypoint: npm 
+    args: ['ci']
+
+  - name: node
```
Comment (Jezzamonn) [inline .cloudbuild/staging-deploy.yaml:16]: Any reason this is the "node" image?

**Review by Jezzamonn — APPROVED** [review]
No obvious issues to me, although I am not particularly fluent with cloud build :)

File: .cloudbuild/staging-deploy.yaml (hunk 2 above)
Comment (Jezzamonn) [reply .cloudbuild/staging-deploy.yaml:16]: Er, nevermind. I checked the contents of the staging-deploy script.

File: .cloudbuild/staging/appengine/http.go (hunk 1 above)
Comment (samthor) [reply .cloudbuild/staging/appengine/http.go]: good call. The internal elfbuilder version of this used this part for whitelist code (which I've removed).
But turns out if everything is public in the repo, then the fact that we stage a test version of Santa is not very secret either...

**Pull request merged**